src/kagglehub/integrity.py

```diff
@@ -0,0 +1,36 @@
+import base64
+from typing import Optional
+
+import requests
+
+# See https://cloud.google.com/storage/docs/xml-api/reference-headers#xgooghash
+GCS_HASH_HEADER = "x-goog-hash"
+COMPUTE_HASH_CHUNK_SIZE = 8192
+
+
+def get_md5_checksum_from_response(response: requests.Response) -> Optional[str]:
+    # See https://cloud.google.com/storage/docs/xml-api/reference-headers#xgooghash
+    # Format is: x-goog-hash: crc32c=n03x6A==,md5=Ojk9c3dhfxgoKVVHYwFbHQ==
+    if GCS_HASH_HEADER in response.headers:
+        header_value = response.headers[GCS_HASH_HEADER]
+        for checksum in header_value.split(","):
```

Comment:
Should you add a file test_integrity.py? I manually tested the case `x-goog-hash: crc32c=n03x6A==,md5=Ojk9c3dhfxgoKVVHYwFbHQ==` and it's working, but it may also help to debug a failing test in the future.